iroh/src/services.rs

```diff
@@ -1,10 +1,124 @@
-use std::io::{stdout, Write};
-
-use anyhow::Result;
+use anyhow::{anyhow, Result};
 use crossterm::terminal::{Clear, ClearType};
 use crossterm::{cursor, style, style::Stylize, QueueableCommand};
 use futures::StreamExt;
+use iroh_util::iroh_cache_path;
+use std::collections::HashSet;
+use std::io::{stdout, Write};
+use tracing::info;
+
 use iroh_api::{Api, ServiceStatus, StatusRow, StatusTable};
+use iroh_util::lock::{read_lock_pid, try_cleanup_dead_lock, LockError, ProgramLock};
+
+/// start any of {iroh-gateway,iroh-store,iroh-p2p} that aren't currently
+/// running.
+pub async fn start(api: &impl Api) -> Result<()> {
+    // start_services(api, HashSet::from(["store"])).await
+    start_services(api, HashSet::from(["store", "p2p", "gateway"])).await
+}
+
+// TODO(b5) - should check for configuration mismatch between iroh CLI configuration
+async fn start_services(api: &impl Api, services: HashSet<&str>) -> Result<()> {
+    // check for any running iroh services
+    let table = api.check().await;
+
+    let mut missing_services = HashSet::new();
+    let missing_services = table
+        .iter()
+        .fold(&mut missing_services, |accum, status_row| {
```

Comment:
This may be lack of Rust experience, but this use of `fold` surprised me as I'd have expected the accumulator to be treated as immutable. Wouldn't a simple `for` loop be more easy to read for what's really repeated mutation of `missing_services`?